platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPlugin.java

```diff
@@ -228,6 +230,10 @@ private void addDefaultReportTask(final JacocoPluginExtension extension, final T
                 reportTask.setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
                 reportTask.setDescription(String.format("Generates code coverage report for the %s task.", testTaskName));
                 reportTask.executionData(testTaskProvider.get());
+                String compileJavaEncoding = compileJavaTaskProvider.get().getOptions().getEncoding();
```

Comment:
🤔 This wiring (and in the coverage verification task) should probably use Property mapping like:

```
reportTask.getSourceEncoding().set(project.provider(() -> compileJavaTaskProvider.get().getOptions().getEncoding()).orElse(Charset.defaultCharset().name()));
```

Reply:
Definitely, already changed that. :-)